metrics.md

```diff
@@ -30,6 +30,55 @@ The following are the minimal metrics agreed to be conformed by the various clie
+| `beacon_custody_columns_count_total`                              | Counter     | Total count of columns in custody within the data availability boundary                                     | On custody collecting and verification |
+
+#### Gossipsub metrics
+Gossipsub domain metrics should contain topic `data_column_sidecar_{subnet_id}` as labels.
+
+| Name | Metric type | Usage | Sample collection event |
+|--------------------------------------------|-------------|-------------------------------------------------------------|----------------------|
+| `gossipsub_topic_msg_sent_counts_total` | Counter | Number of gossip messages sent to each topic | On sending a message over a topic |
+| `gossipsub_topic_msg_sent_bytes_total` | Counter | Number of bytes sent to each topic | On sending a message over a topic |
+| `gossipsub_topic_msg_recv_counts_unfiltered_total` | Counter | Number of gossip messages received from each topic (including duplicates)  | On receiving a message from a topic including duplicates|
+| `gossipsub_topic_msg_recv_bytes_unfiltered_total` | Counter | Number of bytes received from each topic (including duplicates) |  On receiving a message from a topic including duplicates |
+| `gossipsub_topic_msg_recv_counts_total` | Counter | Number of gossip messages received from each topic (deduplicated)  | On receiving a message from a topic deduplicated |
+| `gossipsub_topic_msg_recv_bytes_total` | Counter | Number of bytes received from each topic (deduplicated) | On receiving a message from a topic deduplicated |
+
+
+#### Req/Resp metrics 
+
+Req/Resp domain metrics should contain protocol ID `/eth2/beacon_chain/req/data_column_sidecars_by_root/1/` and `/eth2/beacon_chain/req/data_column_sidecars_by_range/1/` as labels.
+
+| Name | Metric type | Usage | Sample collection event |
+|--------------------------------------------|-------------|-------------------------------------------------------------|----------------------|
+| `libp2p_rpc_requests_sent_total` | Counter | Number of requests sent | On propagating an RPC request |
+| `libp2p_rpc_requests_bytes_sent_total` | Counter | Number of requests bytes sent | On propagating an RPC request |
+| `libp2p_rpc_requests_received_total` | Counter | Number of requests received |  On receiving an RPC request  |
+| `libp2p_rpc_requests_bytes_received_total` | Counter | Number of requests bytes received | On receiving an RPC request |
+| `libp2p_rpc_responses_sent_total` | Counter | Number of responses sent  | On propagating an RPC response |
+| `libp2p_rpc_responses_bytes_sent_total` | Counter | Number of responses bytes sent  | On propagating an RPC response |
+| `libp2p_rpc_responses_received_total` | Counter | Number of responses received  | On receiving an RPC response |
+| `libp2p_rpc_responses_bytes_received_total` | Counter | Number of responses bytes received | On receiving an RPC response |
```

Comment:
How do you feel about prefixing this with `req_resp` instead of `libp2p_rpc`?

Reply:
@parithosh

Reply:
is there a specific reason @matthewkeil ?

Reply:
Mostly because the domains prescribed in the spec are "gossipsub" and "req/resp" so thinking that domain specific metrics should follow that convention (similar to how the `gossipsub` are labeled in this PR).

Both transmission methodologies are `p2p` so not sure that is a necessary differentiator.  Also, at least within our tech stack, both "gossipsub" and "req/resp" are using `@libp2p` so qualifying one with that prefix and not the other doesnt seem correct (also seems redundant and implementation specific).

Reply:
As a newcomer, I found it confusing that specs refer to req/resp, while metrics use a libp2p prefix. Needed extra explanation.

Reply:
The terms `rpc` & `req/resp` have been used interchangeably throughout the implementation and discussions. I feel that `req/resp` is a bit more precise, though `rpc` is more commonly used outside of the spec.

In Lighthouse, most of our metrics and code references `rpc`, and I'd personally prefer to keep it consistent there and also avoid breaking changes to existing metrics. However I think we can probably find ways around this, e.g. metrics relabelling on export (will use more storage) or configure mapping in Grafana, if standardising these metrics provide benefits.